### PR TITLE
Add new component `LayerOpacityPicker.vue` (ref: legend item context menu)

### LINKED ISSUE #353: Change Opacity Layer
### Checklist

- [X] I've searched through the [current issues](https://github.com/g3w-suite/g3w-client/issues) to make sure this feature hasn't been requested already.

### Motivation

It could be useful change opacity of a layer to show element on map cover by a layer features.

### Suggested solution

Add a item to Layer Context Menu on TOC to change layer opacity

### Alternatives considered

_No response_

### PR DESCRIPTION
**Pull request by volterra79**
Closes: #353

Add an item on layer context menu (right click on layer that has geometry) to change layer opacity as follow image:

Screenshot from 2023-02-21 11-17-22

### CODE REVIEW

File: src/components/CatalogLayerContextMenu.vue [hunk 1]
```diff
@@ -32,7 +32,43 @@
...
+            :sync="false"
+            :showValue="true"
+            :unit="'%'"
+            @changed="closeLayerMenu(layerMenu.layerOpacityMenu)"
+            @change-range="setLayerOpacity">
+          </range>
+        </li>
+      </ul>
+    </li>
+
```
Comment (Raruto) [inline src/components/CatalogLayerContextMenu.vue]: Find out how to extract this section into a new separate single file component

Comment (volterra79) [reply src/components/CatalogLayerContextMenu.vue]: @Raruto ok

File: src/components/CatalogLayerContextMenu.vue [hunk 2]
```diff
@@ -32,15 +41,30 @@
...
       </ul>
     </li>
-    <li v-if="canZoom(layerMenu.layer)" @click.prevent.stop="zoomToLayer(layerMenu.layer)">
+
+    <!-- TODO add item description -->
+    <li v-if="layerMenu.layer.geolayer && layerMenu.layer.visible" class="menu-icon">
+      <layer-opacity-picker
+        @add-layer-menu-item="addLayerMenuItem"
+        @show-layer-menu="showSubMenuContext"
+        :layerMenu="layerMenu"
```
Comment (Raruto) [inline src/components/CatalogLayerContextMenu.vue]: replace the `layerMenu` property with the individual (and only those) properties needed by the component `layer-opacity-picker` to work (eg: `layer`, `opacity`, `show`, ... )

File: src/components/LayerOpacityPicker.vue [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+  @file
+  @since v3.8.0
+-->
+
+<template>
+
+  <div
+    @mouseleave.self="showLayerOpacityMenu(false, $event)"
+    @mouseover.self="showLayerOpacityMenu(true, $event)"
+  >
```
Comment (Raruto) [inline src/components/LayerOpacityPicker.vue:11]: Perhaps the sub-context menu positioning should be handled within the `CatalogLayerContextMenu.vue` component. But for now I think it's okay too.

**Review by Raruto — APPROVED** [review]

File: src/components/LayerOpacityPicker.vue (hunk 3 above)
Comment (volterra79) [reply src/components/LayerOpacityPicker.vue:11]: @Raruto i agree. Maybe when we refactor all CatalogLayerComponentMenu in an isolate vue componet we handle it

**Review by Raruto — APPROVED** [review]

**Review by Raruto — APPROVED** [review]
Hoping that the third time will be the good one 🤞

**Pull request merged**